lib/src/sdk_env.dart

```diff
@@ -179,6 +179,7 @@ class ToolEnvironment {
       return output;
     } finally {
       await customOptionsFile.delete();
+      await pedanticOptionsFile.delete();
```

Comment:
How about refactoring this as:
```dart
await withTempFile(name, content, () async {
  ...
});
```

Then have `withTempFile` write the file and delete it after the function given as last parameter is finished...
That way you can easily have both files in their own `finally {` block... which ensures that the second file is deleted, even if the first file wasn't deleted...

side note: couldn't we put these files in `/tmp/`, ideally using https://api.dartlang.org/stable/2.2.0/dart-io/Directory/systemTemp.html

Reply:
Just a suggestion :)

Reply:
I've added a `TODO` to address this later.